src/vs/editor/common/commands/trimTrailingWhitespaceCommand.ts

```diff
@@ -15,13 +15,20 @@ export class TrimTrailingWhitespaceCommand implements editorCommon.ICommand {
 
 	private selection: Selection;
 	private selectionId: string;
+	private preserveCursor: boolean;
 
-	constructor(selection: Selection) {
+	constructor(selection: Selection, preserveCursor?: boolean) {
```

Comment:
Consider `preserveCursor = false` here for a default value (makes it conditional).